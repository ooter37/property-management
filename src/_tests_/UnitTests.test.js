import React from 'react'
import { unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import axios from 'axios'
import renderer from 'react-test-renderer';
import { HashRouter, Link } from 'react-router-dom';


describe('Checking endpoints return unauthorized when user not logged in.', () => {
  it('Get contractors.', () => {
    axios.get('http://localhost:4244/api/contractors').then((res) => {
      expect(res).toBe('User not logged in.')
    })
  });
  it('Get houses.', () => {
    axios.get('http://localhost:4244/api/houses').then((res) => {
      expect(res).toBe('User not logged in.')
    })
  });
  it('Get renters.', () => {
    axios.get('http://localhost:4244/api/renters').then((res) => {
      expect(res).toBe('User not logged in.')
    })
  });
  it('Get tasks.', () => {
    axios.get('http://localhost:4244/api/tasks').then((res) => {
      expect(res).toBe('User not logged in.')
    })
  });
  it('Send email.', () => {
    axios.post('/email/multi').then((res) => {
      expect(res).toBe('User not logged in.')
    })
  });
})

describe('Route testing', () => {
  test(" '/' renders login page (landing).", () => {
    const component = renderer.create(
      <HashRouter>
        <Link to='/' />
      </HashRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test(" '/register' renders registration page.'", () => {
    const component = renderer.create(
      <HashRouter>
        <Link to='/register' />
      </HashRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test(" '/main' renders main page.", () => {
    const component = renderer.create(
      <HashRouter>
        <Link to='/main' />
      </HashRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  //Mark 3
  test(" '/add_house' renders add house page.", () => {
    const component = renderer.create(
      <HashRouter>
        <Link to='/add_house' />
      </HashRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  //Mark 4
  test(" '/update_house' renders update house page.", () => {
    const component = renderer.create(
      <HashRouter>
        <Link to='/update_house' />
      </HashRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});
