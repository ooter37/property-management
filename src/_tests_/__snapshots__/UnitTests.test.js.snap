// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route testing  '/' renders login page (landing). 1`] = `
<a
  href="#/"
  onClick={[Function]}
/>
`;

exports[`Route testing  '/add_house' renders add house page. 1`] = `
<a
  href="#/add_house"
  onClick={[Function]}
/>
`;

exports[`Route testing  '/main' renders main page. 1`] = `
<a
  href="#/main"
  onClick={[Function]}
/>
`;

exports[`Route testing  '/register' renders registration page.' 1`] = `
<a
  href="#/register"
  onClick={[Function]}
/>
`;

exports[`Route testing  '/update_house' renders update house page. 1`] = `
<a
  href="#/update_house"
  onClick={[Function]}
/>
`;
